### PR TITLE
Enable support for Cuda enabled builds on Linux

### PR DESCRIPTION
**Pull request by awvwgk**
MNT: Re-rendered with conda-build 3.23.3, conda-smithy 3.22.1, and conda-forge-pinning 2023.02.17.18.17.27

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* ~~Reset the build number to `0` (if the version changed)~~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by loriab** [comment]
I've minimal cuda experience, but this looks reasonable to me. Should we wait until #43 is resolved to avoid exploding the build matrix?

**Comment by awvwgk** [comment]
Well, that doesn't look promising:

```
[15/336] Building CUDA object CMakeFiles/xc.dir/src/gga_c_ft97.c.o
FAILED: CMakeFiles/xc.dir/src/gga_c_ft97.c.o 
$BUILD_PREFIX/bin/nvcc -forward-unknown-to-host-compiler -DHAVE_CBRT -DXC_ENFORCE_FERMI_HOLE_CURVATURE -Dxc_EXPORTS -I$SRC_DIR/src -I$SRC_DIR/build/src -I$SRC_DIR/build --generate-code=arch=compute_60,code=[compute_60,sm_60] --generate-code=arch=compute_70,code=[compute_70,sm_70] --generate-code=arch=compute_80,code=[compute_80,sm_80] -O3 -std=c++17 --compiler-options -fvisibility-inlines-hidden,-std=c++17,-fmessage-length=0,-march=nocona,-mtune=haswell,-ftree-vectorize,-fPIC,-fstack-protector-strong,-fno-plt,-O2,-ffunction-sections,-pipe,-isystem,$PREFIX/include,-fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/libxc-split-6.1.0,-fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix,-isystem,/usr/local/cuda/include -O3 -DNDEBUG -Xcompiler=-fPIC -MD -MT CMakeFiles/xc.dir/src/gga_c_ft97.c.o -MF CMakeFiles/xc.dir/src/gga_c_ft97.c.o.d -x cu -rdc=true -c $SRC_DIR/src/gga_c_ft97.c -o CMakeFiles/xc.dir/src/gga_c_ft97.c.o
nvcc error   : 'ptxas' died due to signal 9 (Kill signal)
```

**Comment by awvwgk** [comment]
When reducing the derivatives to third order the Cuda builds are passing. I guess that is as good as it gets then.

File: .azure-pipelines/azure-pipelines-linux.yml [hunk 1]
```diff
@@ -8,8 +8,20 @@ jobs:
     vmImage: ubuntu-latest
   strategy:
     matrix:
-      linux_64_:
-        CONFIG: linux_64_
+      linux_64_cuda_compiler_version11.0cxx_compiler_version9fortran_compiler_version9:
```
Comment (loriab) [inline .azure-pipelines/azure-pipelines-linux.yml:11]: Is the plain Linux (no cuda, no aarch64) build missing? I don't see it here or in checks, but I'm on a restricted screen.

Comment (awvwgk) [reply .azure-pipelines/azure-pipelines-linux.yml:11]: The plain Linux build now goes under the name `linux_64_cuda_compiler_versionNone*`, similar to the Windows build which is now named `win_64_cuda_compiler_versionNone`.

**Comment by awvwgk** [comment]
Not sure why the Travis build gets an error. Is there a reason to build ppc64le on native arch rather than cross-compile it like aarch64?

**Comment by loriab** [comment]
I don't see clues on the Travis check either. There's no reason I know not to cross-compile, but I don't know the history. OpenMM has some cuda+ppc64le specifics, in case that's helpful https://github.com/conda-forge/openmm-feedstock/blob/main/recipe/meta.yaml#L20

**Comment by awvwgk** [comment]
Cross compiling ppc64le seems to work super good, maybe we can even increase the number of derivatives here. Testing Cuda support for this is something I don't want to do in this PR.

**Comment by awvwgk** [comment]
> Does the cuda version need to be baked into the build string like tensorflow? [conda-forge/tensorflow-feedstock@`0422d85`/recipe/meta.yaml#L400](https://github.com/conda-forge/tensorflow-feedstock/blob/0422d85657d099710a1d69189c70c11811d4506f/recipe/meta.yaml#L400)

I looked around in other packages, not all include the Cuda version. I also struggled a bit with the template syntax to get it working as it is right now, didn't want to copy and paste a complicated template around everywhere.

**Pull request merged**